RFP-01.md

```diff
@@ -0,0 +1,212 @@
+    // --- view functions ---
+    /**
+     * Determine if a candidate:
+     * 1. Has a registered sub-pool; and
+     * 2. The sub-pool has not announced an exit; and
+     * 3. The sub-pool has no overdue bills
+     * While not enforced on-chain, an address for which this call returns `true` is a
+     * solver and is allowed to submit settlement on CoW Protocol through
+     * `SignedSettlement`.
+     * @param candidate The address to check
+     * @return bool if the address should be allowed to submit settlements through
+     * `SignedSettlement`
+     */     
+    function canSolve(address candidate) external view returns (bool);
+    /**
+     * View function for determining the sub-pool for a given solver.
+     * Reverts if the input address has no sub-pool to its name.
+     * @param solver The solver to retrieve the sub-pool for
+     * @return The contract managing the collateral paid by this solver to be
+     * allowed to settle transactions on CoW Protocol
+     */
+    function solverSubPool(address solver) external view returns (SubPool);
+    /**
+     * View function for determining the backend URI for the sub-pool
+     * @return string URI
+     */
+    function backendUri(SubPool pool) external view returns (string memory);
+
+    // --- permissionless functions ---
+    /**
+     * Deploy a `SubPool` for a solver and fund it with the minimum required collateral.
+     * @notice Deposited collateral consists of `amt` of `token` and `cowAmt` of COW.
+     * @param token The nominated yield bearing token to use for collateral
+     * @param amt How much collateral to add to the sub-pool (minimum amount should be
+     * confirmed off-chain before calling this function)
+     * @param cowAmt Amount of COW to add as collateral (the call will revert if this
+     * value is below the minimal collateral needed for COW - acts as spam prevention
+     * and gate-keeps `canSolve`).
+     * @param solver The address of the solver to create a sub-pool for (the call will
+     * revert if the solver already has a sub-pool or is a contract).
+     * @param backendUri The URI to send the batches to the solver
+     * @dev Emits an event `SubPoolCreated` when creating a sub-pool.
+     */
+    function create(
+        IERC20 token,
+        uint256 amt,
+        uint256 cowAmt,
+        address solver,
+        string calldata backendUri
+    ) external;
+
+    // --- cow dao bonding pool authed functions ---
+
+    /**
+     * Set the minimum delay before a solver can claim back the collateral from its
+     * sub-pool
+     * @dev Emits an event when setting a delay
+     * @param delay The delay in seconds
+     */
+    function setExitDelay(uint256 delay) external auth;
+    /**
+     * Set the minimum collateral required for $COW
+     * @dev Emits an event when setting the minimum collateral
+     * @param cowAmt The minimum amount of COW required
+     */
+    function setMinCow(uint256 cowAmt) external auth;
```

Comment:
Is this still needed given we got rid on enforcing concrete bonding amounts?

Reply:
Having this provides some spam prevention and increases cost for some attacks (notably the `canSolve` would be protected by the minimum cow price, giving a determined value of which attacks must exceed to be profitable - i.e. if the off-chain infra was compromised / had security flaws)

Reply:
I don't really see how spam protection plays a role here (I assume the off-chain component tracks a fixed list of addresses it wants to check if they can solve), but in this case I think it should be possible to set min requirements for a bunch of tokens (not just CoW). However, then it becomes a question how `or` relations ships (e.g. 500k COW and 100kUSDC or 100kUSDT) should be modelled, which again adds a lot of complexity, which is why I think we should drop it.

If this is mainly to predict future use of view functions (ie canSolve), how about making this contract inherit from [StorageAccessible](https://github.com/gnosis/util-contracts/blob/main/contracts/storage/StorageAccessible.sol) so we can extend it with view functions later on (and even deploy `canSolve` functions with hardcoded min amounts).

Reply:
It's simply that it doesn't make much sense to be able to call `canSolve(bob)` and it return `true` if bob has just called `create` but transferred nothing.